### PR TITLE
Migration to core22 and other changes

### PR DESCRIPTION
**Pull request by soumyaDghosh**
Kindly check the snap size, if you can reduce it, the same problem is with amberol also. Kindly test if arm64 and armhf is actually building or not, actually it's too late, so, got no time.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -2,13 +2,13 @@ name: fractal
...
+version: 4.4.2
+icon: org.gnome.Fractal.svg
 architectures:
   - build-on: amd64
-
-layout:
-  /usr/share/fractal/resources.gresource:
-    symlink: $SNAP/usr/share/fractal/resources.gresource
+  - build-on: arm64
+  - build-on: armhf
```
Comment (kenvandine) [inline snap/snapcraft.yaml]: Please drop armhf, I'd rather not support that arch at this time

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -19,39 +19,101 @@ slots:
 
 apps:
   fractal:
-    extensions: [gnome-3-38]
+    extensions: [gnome]
     command: usr/bin/fractal
     common-id: org.gnome.Fractal
     desktop: usr/share/applications/org.gnome.Fractal.desktop
     plugs:
-      - audio-playback
```
Comment (kenvandine) [inline snap/snapcraft.yaml:27]: I think we need to keep this audio-playback

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -19,39 +19,101 @@ slots:
...
       - home
-      - gsettings
-      - network
```
Comment (kenvandine) [inline snap/snapcraft.yaml:30]: network should be required

File: snap/snapcraft.yaml [hunk 4]
```diff
@@ -19,39 +19,101 @@ slots:
...
+  gtksourceview:
+    source: https://gitlab.gnome.org/GNOME/gtksourceview.git
+    source-branch: 'gtksourceview-4-6'
+    source-depth: 1
+    plugin: meson
+    meson-parameters:
+     - --prefix=/usr
+     - -Ddebug=true
+     - -Doptimization=3
+    build-packages:
```
Comment (kenvandine) [inline snap/snapcraft.yaml:75]: Are you sure we need these build packages?  These are both included in the sdk snap

File: snap/snapcraft.yaml [hunk 5]
```diff
@@ -19,39 +19,101 @@ slots:
...
-      - appstream
       - curl
       - libssl-dev
       - libgstreamer1.0-dev
       - libgstreamer-plugins-bad1.0-dev
       - libgstreamer-plugins-base1.0-dev
       - libgstreamer-plugins-good1.0-dev
       - libges-1.0-dev
-      - cmake
+      - libhandy-0.0-dev
```
Comment (kenvandine) [inline snap/snapcraft.yaml:117]: libhandy-1 is in the sdk snap... perhaps fractal still needs the older one, just double check what is really needed

File: snap/snapcraft.yaml [hunk 6]
```diff
@@ -62,42 +124,29 @@ parts:
       - libgstreamer-plugins-bad1.0-0
       - libgstreamer-plugins-base1.0-0
       - libgstreamer-plugins-good1.0-0
+      - libhandy-0.0-0
```
Comment (kenvandine) [inline snap/snapcraft.yaml:130]: libhandy-1 is in the sdk snap... perhaps fractal still needs the older one, just double check what is really needed

File: snap/snapcraft.yaml [hunk 7]
```diff
@@ -19,39 +19,101 @@ slots:
...
   fractal:
+    after: [rustup, gtksourceview, ffmpeg]
     parse-info: [usr/share/metainfo/org.gnome.Fractal.metainfo.xml]
     source: https://gitlab.gnome.org/GNOME/fractal.git
     source-type: git
-    source-commit: 8781378a797d10d6b8e22698aa7cc8d825f13ae5
+    source-tag: $SNAPCRAFT_PROJECT_VERSION
     plugin: meson
-    meson-parameters: [--prefix=/snap/fractal/current/usr]
+    meson-parameters: [--prefix=/snap/fractal/current/usr, -Dprofile=default]
```
Comment (kenvandine) [inline snap/snapcraft.yaml]: It would be better to make prefix /usr

File: snap/snapcraft.yaml [hunk 8]
```diff
@@ -19,39 +19,101 @@ slots:
...
-      - audio-playback
       - home
-      - gsettings
-      - network
-      - opengl
-      - password-manager-service
+      - removable-media
 
 parts:
+  ffmpeg:
```
Comment (kenvandine) [inline snap/snapcraft.yaml:37]: Does fractal really use ffmpeg now?  If so, does it still use gstreamer?  Perhaps gstreamer could be removed now...

**Review by kenvandine — CHANGES_REQUESTED** [review]

File: snap/snapcraft.yaml (hunk 1 above)
Comment (soumyaDghosh) [reply snap/snapcraft.yaml]: Okay, no problem. Removing them

File: snap/snapcraft.yaml (hunk 2 above)
Comment (soumyaDghosh) [reply snap/snapcraft.yaml:27]: I removed audio-playback and network by accident ig

File: snap/snapcraft.yaml (hunk 5 above)
Comment (soumyaDghosh) [reply snap/snapcraft.yaml:117]: fractal needs libhandy-0 and libhandy-1 is in gnome-sdk. I'll build the latest master branch today.

File: snap/snapcraft.yaml (hunk 8 above)
Comment (soumyaDghosh) [reply snap/snapcraft.yaml:37]: I got it from the flatpak manifest.

https://gitlab.gnome.org/GNOME/fractal/-/blob/4.4.2/flatpak/org.gnome.Fractal-release.json#L27

File: snap/snapcraft.yaml (hunk 7 above)
Comment (soumyaDghosh) [reply snap/snapcraft.yaml]: The layout will be needed then

**Comment by soumyaDghosh** [comment]
I hope you haven't forgot this. :face_in_clouds:

File: snap/snapcraft.yaml (hunk 7 above)
Comment (kenvandine) [reply snap/snapcraft.yaml]: I would rather have the layout and the correct prefix rather than the incorrect prefix.

**Review by kenvandine — APPROVED** [review]
Thanks for the fixes

**Comment by soumyaDghosh** [comment]
If you have time, kindly check the iotas snap

**Comment by soumyaDghosh** [comment]
I pushed this just before the exam started. LOL

**Pull request merged**